earthmover/earthmover.py

```diff
@@ -121,11 +122,23 @@ def load_project_configs(self, filepath: str):
         """
         configs = JinjaEnvironmentYamlLoader.load_project_configs(filepath, params=self.params)
 
-        # Update project parameter defaults from the template, if any
+        # 1. Update project parameter defaults from the template, if any
         for key, val in configs.get("parameter_defaults", {}).items():
             self.params.setdefault(key, val)
 
-        # Prepend package macros to the project macro string. Later macro definitions in the string will overwrite earlier ones
+        # 2. There may be config keys that expect an environment variable but for which a default is also defined.
+        #     If no env var was passed by the user, apply the default value.
+        env_var_pattern = r'^\$\{[A-Z_]+\}$'
+        regex = re.compile(env_var_pattern)
+
+        for key, val in configs.items():
+            if isinstance(val, str) and regex.match(val):
+                # Combine `key: ${VAL}` with `VAL: default_val` to get `key: default_val`
+                val_as_env_var = val.replace("$", "").replace("{", "").replace("}", "")
+                if val_as_env_var in self.params:
+                    configs[key] = self.params[val_as_env_var]
```

Comment:
Could we use string templates here instead? something like
```python
from string import template

for key, val in configs.items():
    if isinstance(val, str):
        template = Template(val)
        configs[key]= template.substitute(self.params)
```
This would eliminate the need for regex matching, and also support using an env var as _part_ of the config value but not the _whole_ value, like
```yaml
config:
  output_dir: ${OUTPUT_BASE}transformed/
  parameter_defaults:
    OUTPUT_BASE: './output/'
```

Reply:
WOW, dramatically simpler, this is awesome, thanks! Tested that it works like a chjarm